### PR TITLE
Add Global Shadow Props to ALL kits

### PR DESCRIPTION
**Pull request by pauljpark**
#### Screens
***Screenshots here are just for PROOF OF CONCEPT***

Screen Shot 2021-12-02 at 9 50 03 AM

Screen Shot 2021-12-02 at 9 50 38 AM

#### Breaking Changes

[Yes/No (Explain)]

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/PLAY-31

#### How to test this

***FOR REACT***
Pass one of the values `none | deep | deeper | deepest` to `shadow` on a kit `eg. _button_default.jsx` which will look like `eg. shadow="deepest"` within that component. Some kits will override the prop so the shadow style will not be applied to such kits.

***FOR RAILS***
Pass one of the values `none | deep | deeper | deepest` to `shadow` on a kit `eg. _button_default.html.erb` which will look like `eg. shadow: "deepest"` within that component. Some kits will override the prop so the shadow style will not be applied to such kits.

#### Checklist:

- [ ] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [ ] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [ ] **SCREENSHOT** Please add a screen shot or two.
- [ ] **SPECS** Please cover your changes with specs.
- [ ] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

**Review by thestephenmarshall — APPROVED** [review]
Thank you for the attention to detail on this one. Looks great to me! :shipit:

File: playbook/app/pb_kits/playbook/utilities/_shadow.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
+@import "../tokens/exports/colors";
+@import "../tokens/exports/opacity";
+
+$shadow_none: 0 0 0 0 transparent;
+$shadow_deep: 0 4px 10px 0 rgba($shadow, 0.16);
+$shadow_deeper: 0 12px 28px 0 rgba($shadow, 0.18);
+$shadow_deepest: 0 30px 38px 4px $shadow, 0 2px 14px 4px rgba($shadow, 0.1);
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/utilities/_shadow.scss:7]:
```suggestion
@import "../tokens/exports/shadows";
```

No need to redefine this here. It already exists as a token

File: playbook/app/pb_kits/playbook/utilities/_shadow.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  box-shadow: $shadow_deep;
+}
+
+.shadow_deeper { 
+  box-shadow: $shadow_deeper;
+}
+
+.shadow_deepest { 
+  box-shadow: $shadow_deepest; 
+}
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/utilities/_shadow.scss:23]:
```suggestion
@mixin shadow-classes($shadows-list) {
  @each $name, $shadow in $shadows-list {
    .shadow_#{$name} {
      box-shadow: $shadow;
    }
  }
}

@include shadow-classes($box_shadows);
```

Let's make this smarter. We already have an awesome map we can loop over to dynamically generate these.

**Review by jasoncypret — CHANGES_REQUESTED** [review]

**Review by jasoncypret — CHANGES_REQUESTED** [review]
One more: 

image

The shadows are no longer showing on the docs view. Can you investigate and fix?

**Review by jasoncypret — APPROVED** [review]

**Pull request merged**